spikeinterface/preprocessing/interpolate_bad_channels.py

```diff
@@ -0,0 +1,120 @@
+
+    Returns
+    -------
+    interpolated_recording: InterpolateBadChannelsRecording
+        The recording object with interpolated bad channels
+    """
+    name = 'interpolate_bad_channels'
+
+    def __init__(self, recording, bad_channel_ids, sigma_um=None, p=1.3, weights=None):
+        BasePreprocessor.__init__(self, recording)
+
+        bad_channel_ids = np.array(bad_channel_ids)
+        self.check_inputs(recording, bad_channel_ids)
+
+        self.bad_channel_ids = bad_channel_ids
+        self._bad_channel_idxs = recording.ids_to_indices(self.bad_channel_ids)
+        self._bad_channel_idxs.setflags(write=False)
+
+        if sigma_um is None:
+            sigma_um = estimate_recommended_sigma_um(recording)
+
+        if weights is None:
+            weights = preprocessing_tools.get_kriging_channel_weights(recording.get_channel_locations(),
+                                                                      self._bad_channel_idxs,
+                                                                      sigma_um,
+                                                                      p)
+
+        for parent_segment in recording._recording_segments:
+            rec_segment = InterpolateBadChannelsSegment(parent_segment,
+                                                        self._bad_channel_idxs,
+                                                        weights)
+            self.add_recording_segment(rec_segment)
+
+        self._kwargs = dict(recording=recording.to_dict(),
+                            bad_channel_ids=bad_channel_ids,
+                            p=p,
+                            sigma_um=sigma_um,
+                            weights=weights)
+
+    def check_inputs(self, recording, bad_channel_ids):
+
+        if bad_channel_ids.ndim != 1:
+            raise TypeError("'bad_channel_ids' must be a 1d array or list.")
+
+        if recording.get_property('contact_vector') is None:
+            raise ValueError('A probe must be attached to use bad channel interpolation. Use set_probe(...)')
+
+        if recording.get_probe().si_units != "um":
+            raise NotImplementedError("Channel spacing units must be um")
+
+
+class InterpolateBadChannelsSegment(BasePreprocessorSegment):
+
+    def __init__(self, parent_recording_segment, bad_channel_indexes, weights):
+        BasePreprocessorSegment.__init__(self, parent_recording_segment)
+
+        self._bad_channel_indexes = bad_channel_indexes
+        self._weights = weights
+
+    def get_traces(self, start_frame, end_frame, channel_indices):
+        if channel_indices is None:
+            channel_indices = slice(None)
+
+        traces = self.parent_recording_segment.get_traces(start_frame,
+                                                          end_frame,
+                                                          slice(None))
+
+        traces = traces.copy()
+
+        traces[:, self._bad_channel_indexes] = traces @ self._weights
```

Comment:
I think we should have a diffrents approach here.
Something like:
traces_of_good_channels @ self._weights